src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -748,9 +748,9 @@ void registerStorageKafka(StorageFactory & factory)
 
         auto num_consumers = kafka_settings->kafka_num_consumers.value;
 
-        if (num_consumers > 16)
+        if (num_consumers > 48)
```

Comment:
`getNumberOfPhysicalCPUCores()` instead of some const looks better.

Reply:
yes   getNumberOfPhysicalCPUCores() / 2 ?

Reply:
Yes, `getNumberOfPhysicalCPUCores() / 2` looks OK, I think that even `getNumberOfPhysicalCPUCores()` is OK.
Since using more cores than you have CPUs is likely an error, but using half of cores not always an error.

Also this may break compatibility, if clickhouse was run with more consumers than CPUs on machine, but I guess it is fine, since it is likely an error anyway.

Reply:
Also don't forget to update:
- commit subject/description
- pull request subject/description

Reply:
Can this pr merge to v21.3.16 ?

Reply:
I'm not the one who decide this, maybe @alexey-milovidov can comment on this?

Reply:
@azat The CI is finished, and some errors is not cause by this pr. 
Can the pr merge into master branch ?

Reply:
> Can this pr merge to v21.3.16 ? We want to use it in this version.

@alexey-milovidov It's OK ?